### PR TITLE
Make prioirty tuple

### PR DESCRIPTION
**Pull request by Connor1996**
to support scheduling based on group priority

### CODE REVIEW

File: src/queue/priority.rs [hunk 1]
```diff
@@ -70,11 +70,39 @@ where
...
+            .then_with(|| self.value.cmp(&other.value))
+    }
+}
+
 /// A trait used to generate priority value for each task.
 pub trait TaskPriorityProvider: Send + Sync + 'static {
     /// Return a priority value of this task, all tasks in the priority
     /// queue is ordered by this value.
-    fn priority_of(&self, extras: &Extras) -> u64;
+    fn priority_of(&self, extras: &Extras) -> Priority;
```
Comment (glorv) [inline src/queue/priority.rs:105]: Typically, the total number of priorities is not very big, so what about sharing the high bits of the value field, so we can still use a u64 here and keep this trait simple.

Comment (Connor1996) [reply src/queue/priority.rs:105]: I don't think it makes the trait complicated.  A defined struct is more readable and extensible

Comment (glorv) [reply src/queue/priority.rs:105]: But not all case need the `task_priority`, this change some what put implementation details into this trait

File: src/queue/priority.rs [hunk 2]
```diff
@@ -70,11 +70,39 @@ where
...
+    fn default() -> Self {
+        Priority {
+            group_priority: 8,
+            value: 0,
+        }
+    }
+}
+
+impl Ord for Priority {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
```
Comment (BusyJay) [inline src/queue/priority.rs:93]: What's the difference from `#[derive(Ord)]`?

Comment (Connor1996) [reply src/queue/priority.rs:93]: note that `group_prioirty` is in descending order

**Comment by Connor1996** [comment]
Close it as we plan to do it TiKV by encoding the priority as a u64

**Pull request closed without merging**